### DISCUSSION TITLE
For loop in flows

### DISCUSSION THREAD

**Discussion by Attacler** [directus/directus#14508] [Feature Requests]
At the moment Flows are not supporting loops.
Implementing this can be useful for example for looping trough items or API result(s).

Usecase 1:
When someone updates an invoice lineitem (custom collection with relation to invoice), then loop trough all the line items to calculate the total invoice amount.

Usecase 2:
Loop trough all quotes which are not accepted yet but are after the expiration date and send a reminder to the customer.

**Comment by rijkvanzanten** [comment]
This should be done through the `trigger` operation, as adding inline loops removes the linearity of the workflow. The current `trigger` operation will already fire the related flow once for each submitted payload item if the passed payload is an array rather than an object. Those are ran in parallel and should tackle use case 2.

For use case 1, that should be a _reduce_ operation, rather than a loop. You're trying to generate a single new value from an array of other values 👍🏻

**Reply by rijkvanzanten** [reply]
I also think use case 1 would be covered by #13944 perfectly 🙂

**Reply by Attacler** [reply]
@rijkvanzanten currently im using a CRON for the second point so it will read Data and read data will return all invoices.
Does trigger accept an array and loop trough it by itself? (also it will be messy if you have to create flows for flows just because you cannot loop)

**Reply by rijkvanzanten** [reply]
> Does trigger accept an array and loop trough it by itself

Yes

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
